### PR TITLE
Add failing test for edge case in #1041

### PR DESCRIPTION
**Pull request by evocateur**
As noted in #1043, related to changes originally made in #1015.

### CODE REVIEW

File: test/validation.js [hunk 1]
```diff
@@ -475,6 +425,69 @@ describe('validation tests', () => {
     })
   })
 
+  describe('requiresArg', () => {
+    it('fails when a required argument value of type number is missing', (done) => {
+      yargs()
+        .option('w', {type: 'number', requiresArg: true})
+        .parse('-w', (err, argv, output) => {
```
Comment (evocateur) [inline test/validation.js]: I also changed all of these tests to use `parse(...)` instead of `fail().argv` because the `fail()` pattern takes 8 seconds to error when it fails, vs instantly in the `parse()` pattern.

**Comment by bcoe** [comment]
@evocateur I believe we pulled these tests in in your other branch?

**Comment by evocateur** [comment]
Yep!
